cpp/modmesh/CMakeLists.txt

```diff
@@ -49,10 +49,16 @@ set(MODMESH_ROOT_HEADERS
     ${CMAKE_CURRENT_SOURCE_DIR}/base.hpp
     ${CMAKE_CURRENT_SOURCE_DIR}/math.hpp
     ${CMAKE_CURRENT_SOURCE_DIR}/grid.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/SerializableItem.hpp
+    CACHE FILEPATH "" FORCE)
+
+set(MODMESH_ROOT_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/SerializableItem.cpp
```

Comment:
An implementation file is like an implementation detail and I'd like it to be in a sub-directory instead of the root.

There is not an implementation file in the root, yet.

Reply:
sure, moved the serialization code to the `serialization` directory.